File: tara-admin-ui/src/app/alerts/model/alert.ts
export type Alert = {
  id?: string,
  title: string,
  start_time: string,
  end_time: string,
  login_alert: LoginAlert,
  email_alert?: EmailAlert,
  created_at?: string,
  updated_at?: string
}

export type LoginAlert = {
  enabled: boolean,
  auth_methods: string[],
  message_templates: MessageTemplate[]
}

export type EmailAlert = {
  enabled: boolean,
  send_at: string,
  message_templates: MessageTemplate[]
}

export type MessageTemplate = {
  message: string,
  locale: string
}

File: tara-admin-api/src/main/java/ee/ria/tara/repository/model/Alert.java
package ee.ria.tara.repository.model;

import com.fasterxml.jackson.databind.JsonNode;
import ee.ria.tara.repository.helper.JsonConverter;
import ee.ria.tara.repository.helper.StringListConverter;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.Column;
import jakarta.persistence.Convert;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import java.time.OffsetDateTime;
import java.util.List;

@Entity
@Setter
@Getter
@ToString
@EqualsAndHashCode
public class Alert {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private OffsetDateTime startTime;
    private OffsetDateTime endTime;
    private Boolean notifyClientsOnTaraLoginPage;
    @Convert(converter = JsonConverter.class)
    private JsonNode notificationTemplates;
    @Convert(converter = StringListConverter.class)
    private List<String> displayOnlyForAuthmethods;
    private Boolean notifyClientsByEmail;
    private OffsetDateTime sendAt;
    private String emailTemplate;
    @CreationTimestamp
    @Column(updatable = false)
    private OffsetDateTime createdAt;
    @UpdateTimestamp
    private OffsetDateTime updatedAt;
}
